File: android/app/src/main/java/com/laneshadow/ui/components/molecules/Separator.kt
package com.laneshadow.ui.components.molecules

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.fillMaxHeight
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.width
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.semantics.contentDescription
import androidx.compose.ui.semantics.semantics
import androidx.compose.ui.unit.dp
import com.laneshadow.theme.LocalLaneShadowTheme

/**
 * Separator orientation enum
 *
 * Following RN wrapper API from react-native/components/ui/separator.tsx
 */
enum class SeparatorOrientation {
    Horizontal,
    Vertical,
}

/**
 * Separator molecule component
 *
 * Visual divider line with semantic theme styling.
 * Following React Native wrapper patterns from react-native/components/ui/separator.tsx
 *
 * Horizontal: 1dp height, full width, color from border.default
 * Vertical: 1dp width, full height, color from border.default
 *
 * @param orientation Separator orientation (horizontal or vertical, default: horizontal)
 * @param modifier Modifier for the separator
 */
@Composable
fun Separator(
    orientation: SeparatorOrientation = SeparatorOrientation.Horizontal,
    modifier: Modifier = Modifier,
) {
    val theme = LocalLaneShadowTheme.current

    when (orientation) {
        SeparatorOrientation.Horizontal -> {
            Box(
                modifier = modifier
                    .fillMaxWidth()
                    .height(1.dp)
                    .background(theme.colors.border.default)
                    .semantics {
                        contentDescription = "Separator"
                    },
            )
        }
        SeparatorOrientation.Vertical -> {
            Box(
                modifier = modifier
                    .width(1.dp)
                    .fillMaxHeight()
                    .background(theme.colors.border.default)
                    .semantics {
                        contentDescription = "Separator"
                    },
            )
        }
    }
}

File: ios/LaneShadow/Views/Molecules/Separator.swift
import LaneShadowTheme
import SwiftUI

// MARK: - Separator Orientation

/**
 * Separator orientation enum
 *
 * Defines the layout direction for the separator line.
 */
public enum LSSeparatorOrientation: Sendable {
    case horizontal
    case vertical
}

// MARK: - Separator Component

/**
 * Separator molecule component
 *
 * Visual divider line with semantic theme styling.
 * Following React Native component from react-native/components/ui/separator.tsx
 *
 * ## Design Tokens Used
 * - Colors: `theme.colors.border.default` for the line color
 * - Layout:
 *   - Horizontal: height 1, width .infinity
 *   - Vertical: width 1, height .infinity
 *
 * ## Parameters
 * - orientation: Layout direction (horizontal or vertical, default: horizontal)
 */
public struct LSSeparator: View {
    @Environment(\.theme) private var theme

    private let orientation: LSSeparatorOrientation

    /// Creates a Separator
    /// - Parameters:
    ///   - orientation: Layout direction (default: horizontal)
    public init(
        orientation: LSSeparatorOrientation = .horizontal
    ) {
        self.orientation = orientation
    }

    public var body: some View {
        Rectangle()
            .fill(theme.colors.border.default)
            .frame(
                width: orientation == .vertical ? 1 : nil,
                height: orientation == .horizontal ? 1 : nil
            )
            .frame(
                maxWidth: orientation == .horizontal ? .infinity : nil,
                maxHeight: orientation == .vertical ? .infinity : nil
            )
            .accessibilityLabel("Separator")
    }
}

// MARK: - Preview

#Preview("Horizontal Separator") {
    VStack(spacing: 16) {
        Text("Content above separator")
            .font(.body)
        LSSeparator(orientation: .horizontal)
        Text("Content below separator")
            .font(.body)
    }
    .laneShadowTheme()
    .padding()
}

#Preview("Vertical Separator") {
    HStack(spacing: 16) {
        Text("Left content")
            .font(.body)
        LSSeparator(orientation: .vertical)
        Text("Right content")
            .font(.body)
    }
    .laneShadowTheme()
    .padding()
    .frame(height: 100)
}

#Preview("Multiple Horizontal Separators") {
    VStack(spacing: 0) {
        Text("Section 1")
            .font(.headline)
            .padding()
        LSSeparator(orientation: .horizontal)
        Text("Section 2")
            .font(.headline)
            .padding()
        LSSeparator(orientation: .horizontal)
        Text("Section 3")
            .font(.headline)
            .padding()
    }
    .laneShadowTheme()
}

#Preview("Multiple Vertical Separators") {
    HStack(spacing: 0) {
        Text("A")
            .font(.headline)
            .frame(maxWidth: .infinity)
        LSSeparator(orientation: .vertical)
        Text("B")
            .font(.headline)
            .frame(maxWidth: .infinity)
        LSSeparator(orientation: .vertical)
        Text("C")
            .font(.headline)
            .frame(maxWidth: .infinity)
    }
    .laneShadowTheme()
    .frame(height: 60)
}

#Preview("Dark Theme - Horizontal") {
    VStack(spacing: 16) {
        Text("Dark theme content")
            .font(.body)
        LSSeparator(orientation: .horizontal)
        Text("More dark theme content")
            .font(.body)
    }
    .laneShadowTheme()
    .preferredColorScheme(.dark)
    .padding()
}

#Preview("Dark Theme - Vertical") {
    HStack(spacing: 16) {
        Text("Dark")
            .font(.body)
        LSSeparator(orientation: .vertical)
        Text("Theme")
            .font(.body)
    }
    .laneShadowTheme()
    .preferredColorScheme(.dark)
    .padding()
    .frame(height: 80)
}
